src/apps/jsonld_converter/service/base.py

```diff
@@ -70,6 +70,8 @@ def str_to_id(name: str, to_underscore=r"\s") -> str:
         return ""
     name = re.sub(r"[^0-9a-zA-Z\s_-]+", "", name)
     if to_underscore:
-        name = re.sub(rf"[_{to_underscore}]+", "_", name)
-
+        name = re.sub(rf"_{to_underscore}+", "_", name)
```

Comment:
Please replace regexp with
```
name = re.sub(rf"[_{to_underscore}]+", "_", name)
```
That was a typo)

Reply:
added brackets, so your re.sub is not required in this case

Reply:
So do we move it back to itself? 
`name = re.sub(rf"[_{to_underscore}]+", "_", name)` basically, this regex was removing _ if there was multiple _. Like Activity__1 changed to Activity_1

Reply:
oh sorry, that was the initial variant, I missed that.
So yes, here removed all multiple __.